dec/decode.go

```diff
@@ -122,7 +127,7 @@ func (r *BrotliReader) Read(p []byte) (cnt int, err error) {
 				&availableOut,
 				&nextOut,
 				&r.totalOut,
-				r.state,
+				(*C.BrotliState)(unsafe.Pointer(&r.state[0])),
```

Comment:
:+1: I can't see any other way of handling it. I think unsafe pointers are intentionally long winded to use